### PR TITLE
Use `slog` in envtool

### PR DESCRIPTION
**Pull request by chilagrow**
# Description

Closes #4013.

## Readiness checklist

- [ ] I added/updated unit tests (and they pass).
- [ ] I added/updated integration/compatibility tests (and they pass).
- [ ] I added/updated comments and checked rendering.
- [x] I made spot refactorings.
- [ ] I updated user documentation.
- [ ] I ran `task all`, and it passed.
- [x] I ensured that PR title is good enough for the changelog.
- [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Milestone (`Next`), Labels, Project and project's Sprint fields.
- [x] I marked all done items in this checklist.

### CODE REVIEW

**Comment by chilagrow** [comment]
Here's how integration test output looks like.
```
$ task test-integration-postgresql TEST_RUN=TestInsertFind
task: [gen-version] go generate -x ./build/version
go run ./generate.go
package.txt: build-host                                                                                                                                                                                     
commit.txt: 0acd4ec75736d311a46ad3b72d68be6efc826ebf                                                                                                                                                        
branch.txt: slog-envtool                                                                                                                                                                                    
version.txt: v1.21.0-103-g0acd4ec75                                                                                                                                                                         
task: [gen-version] go build -v -o bin/ -race=true -tags=ferretdb_debug,ferretdb_hana ./cmd/envtool/                                                                                                        
github.com/FerretDB/FerretDB/build/version                                                                                                                                                                  
github.com/FerretDB/FerretDB/internal/util/state                                                                                                                                                            
github.com/FerretDB/FerretDB/cmd/envtool                                                                                                                                                                    
task: [test-integration-postgresql] ../bin/envtool tests run --shard-index=0 --shard-total=0 --run='TestInsertFind' --skip='' -- -count=1 -timeout=35m -race=true -tags=ferretdb_debug,ferretdb_hana -shuffl
e=on -coverpkg=../... -coverprofile=integration-postgresql.txt ./... -target-backend=ferretdb-postgresql -target-tls -postgresql-url='postgres://username@127.0.0.1:5432/ferretdb?search_path=' -compat-url=
'mongodb://username:password@127.0.0.1:47018/?tls=true&tlsCertificateKeyFile=client.pem&tlsCaFile=rootCA-cert.pem&replicaSet=rs0' -disable-pushdown=false
                                                                                                                                                                                                            
INFO    Running go test -list=.TestInsertFind ./...                                                                                                                                                         
INFO    Starting OTLP tracer...                                                                                                                                                                             
INFO    Running go test -json -run=^()$ -count=1 -timeout=35m -race=true -tags=ferretdb_debug,ferretdb_hana -shuffle=on -coverpkg=../... -coverprofile=integration-postgresql.txt ./... -target-backend=ferr
etdb-postgresql -target-tls -postgresql-url=postgres://username@127.0.0.1:5432/ferretdb?search_path= -compat-url=mongodb://username:password@127.0.0.1:47018/?tls=true&tlsCertificateKeyFile=client.pem&tlsC
aFile=rootCA-cert.pem&replicaSet=rs0 -disable-pushdown=false                                                                                                                                                
INFO            github.com/FerretDB/FerretDB/integration/setup          coverage: 0.0% of statements                                                                                                        
INFO            github.com/FerretDB/FerretDB/integration/shareddata             coverage: 0.0% of statements
INFO    PASS github.com/FerretDB/FerretDB/integration/setup                                                                                                                                                 INFO    PASS github.com/FerretDB/FerretDB/integration/shareddata                                                                                                                                            
INFO    2024-07-22T12:34:10.414+0900    INFO    observability/observability.go:89       Starting OTLP tracer... {"name":"otel"}
INFO    2024-07-22T12:34:10.414+0900    INFO    debug   debug/debug.go:221      Starting debug server on http://127.0.0.1:36707...
INFO    2024-07-22T12:34:10.414+0900    INFO    setup/startup.go:135    Target system: ferretdb-postgresql (built-in).
INFO    2024-07-22T12:34:10.415+0900    INFO    debug   debug/debug.go:227      http://127.0.0.1:36707/debug/graphs - Visualize metrics                                                                     INFO    2024-07-22T12:34:10.415+0900    INFO    debug   debug/debug.go:227      http://127.0.0.1:36707/debug/livez - Liveness probe                                                
INFO    2024-07-22T12:34:10.415+0900    INFO    debug   debug/debug.go:227      http://127.0.0.1:36707/debug/metrics - Metrics in Prometheus format
INFO    2024-07-22T12:34:10.415+0900    INFO    debug   debug/debug.go:227      http://127.0.0.1:36707/debug/pprof - Runtime profiling data for pprof
INFO    2024-07-22T12:34:10.415+0900    INFO    debug   debug/debug.go:227      http://127.0.0.1:36707/debug/readyz - Readiness probe
INFO    2024-07-22T12:34:10.415+0900    INFO    debug   debug/debug.go:227      http://127.0.0.1:36707/debug/vars - Expvar package metrics
INFO    2024-07-22T12:34:10.417+0900    INFO    setup/startup.go:153    Compat system: MongoDB (mongodb://username:password@127.0.0.1:47018/?replicaSet=rs0&tls=true&tlsCaFile=%2Fhome%2Fchi%2Fgo%2Fsrc%2Fgi
thub.com%2Fchilagrow%2FFerretDB%2Fbuild%2Fcerts%2FrootCA-cert.pem&tlsCertificateKeyFile=%2Fhome%2Fchi%2Fgo%2Fsrc%2Fgithub.com%2Fchilagrow%2FFerretDB%2Fbuild%2Fcerts%2Fclient.pem).
INFO    -test.shuffle 1721619250417876881                                                             
INFO    testing: warning: no tests to run                                                                                                                                                                   
INFO    PASS                                                                                          
INFO    2024-07-22T12:34:10.422+0900    INFO    observability/observability.go:89       Starting OTLP tracer... {"name":"otel"}
INFO    2024-07-22T12:34:10.423+0900    INFO    setup/startup.go:135    Target system: ferretdb-postgresql (built-in).
INFO    2024-07-22T12:34:10.423+0900    INFO    debug   debug/debug.go:221      Starting debug server on http://127.0.0.1:38505...
INFO    2024-07-22T12:34:10.423+0900    INFO    debug   debug/debug.go:227      http://127.0.0.1:38505/debug/graphs - Visualize metrics
INFO    2024-07-22T12:34:10.423+0900    INFO    debug   debug/debug.go:227      http://127.0.0.1:38505/debug/livez - Liveness probe
INFO    2024-07-22T12:34:10.423+0900    INFO    debug   debug/debug.go:227      http://127.0.0.1:38505/debug/metrics - Metrics in Prometheus format
INFO    2024-07-22T12:34:10.423+0900    INFO    debug   debug/debug.go:227      http://127.0.0.1:38505/debug/pprof - Runtime profiling data for pprof
INFO    2024-07-22T12:34:10.423+0900    INFO    debug   debug/debug.go:227      http://127.0.0.1:38505/debug/readyz - Readiness probe
INFO    2024-07-22T12:34:10.423+0900    INFO    debug   debug/debug.go:227      http://127.0.0.1:38505/debug/vars - Expvar package metrics
INFO    2024-07-22T12:34:10.424+0900    INFO    setup/startup.go:153    Compat system: MongoDB (mongodb://username:password@127.0.0.1:47018/?replicaSet=rs0&tls=true&tlsCaFile=%2Fhome%2Fchi%2Fgo%2Fsrc%2Fgi
thub.com%2Fchilagrow%2FFerretDB%2Fbuild%2Fcerts%2FrootCA-cert.pem&tlsCertificateKeyFile=%2Fhome%2Fchi%2Fgo%2Fsrc%2Fgithub.com%2Fchilagrow%2FFerretDB%2Fbuild%2Fcerts%2Fclient.pem).
INFO    -test.shuffle 1721619250424883750                                                                                                                                                                   
INFO    testing: warning: no tests to run                                                             
INFO    PASS               
INFO    2024-07-22T12:34:10.429+0900    INFO    observability/observability.go:89       Starting OTLP tracer... {"name":"otel"}
INFO    2024-07-22T12:34:10.430+0900    INFO    setup/startup.go:135    Target system: ferretdb-postgresql (built-in).
INFO    2024-07-22T12:34:10.429+0900    INFO    observability/observability.go:89       Starting OTLP tracer... {"name":"otel"}
INFO    2024-07-22T12:34:10.430+0900    INFO    setup/startup.go:135    Target system: ferretdb-postgresql (built-in).
INFO    2024-07-22T12:34:10.430+0900    INFO    debug   debug/debug.go:221      Starting debug server on http://127.0.0.1:46067...
INFO    2024-07-22T12:34:10.430+0900    INFO    debug   debug/debug.go:221      Starting debug server on http://127.0.0.1:39397...
INFO    2024-07-22T12:34:10.430+0900    INFO    debug   debug/debug.go:227      http://127.0.0.1:39397/debug/graphs - Visualize metrics
INFO    2024-07-22T12:34:10.430+0900    INFO    debug   debug/debug.go:227      http://127.0.0.1:46067/debug/graphs - Visualize metrics
INFO    2024-07-22T12:34:10.430+0900    INFO    debug   debug/debug.go:227      http://127.0.0.1:39397/debug/livez - Liveness probe
INFO    2024-07-22T12:34:10.430+0900    INFO    debug   debug/debug.go:227      http://127.0.0.1:46067/debug/livez - Liveness probe
INFO    2024-07-22T12:34:10.430+0900    INFO    debug   debug/debug.go:227      http://127.0.0.1:39397/debug/metrics - Metrics in Prometheus format
INFO    2024-07-22T12:34:10.430+0900    INFO    debug   debug/debug.go:227      http://127.0.0.1:39397/debug/pprof - Runtime profiling data for pprof
INFO    2024-07-22T12:34:10.430+0900    INFO    debug   debug/debug.go:227      http://127.0.0.1:46067/debug/metrics - Metrics in Prometheus format
INFO    2024-07-22T12:34:10.430+0900    INFO    debug   debug/debug.go:227      http://127.0.0.1:39397/debug/readyz - Readiness probe
INFO    2024-07-22T12:34:10.430+0900    INFO    debug   debug/debug.go:227      http://127.0.0.1:46067/debug/pprof - Runtime profiling data for pprof
INFO    2024-07-22T12:34:10.430+0900    INFO    debug   debug/debug.go:227      http://127.0.0.1:39397/debug/vars - Expvar package metrics
INFO    2024-07-22T12:34:10.430+0900    INFO    debug   debug/debug.go:227      http://127.0.0.1:46067/debug/readyz - Readiness probe
INFO    2024-07-22T12:34:10.430+0900    INFO    debug   debug/debug.go:227      http://127.0.0.1:46067/debug/vars - Expvar package metrics
INFO    coverage: 2.0% of statements in ../...
INFO    2024-07-22T12:34:10.432+0900    INFO    setup/startup.go:153    Compat system: MongoDB (mongodb://username:password@127.0.0.1:47018/?replicaSet=rs0&tls=true&tlsCaFile=%2Fhome%2Fchi%2Fgo%2Fsrc%2Fgi
thub.com%2Fchilagrow%2FFerretDB%2Fbuild%2Fcerts%2FrootCA-cert.pem&tlsCertificateKeyFile=%2Fhome%2Fchi%2Fgo%2Fsrc%2Fgithub.com%2Fchilagrow%2FFerretDB%2Fbuild%2Fcerts%2Fclient.pem).
INFO    -test.shuffle 1721619250432160678
INFO    testing: warning: no tests to run
INFO    PASS
INFO    2024-07-22T12:34:10.432+0900    INFO    setup/startup.go:153    Compat system: MongoDB (mongodb://username:password@127.0.0.1:47018/?replicaSet=rs0&tls=true&tlsCaFile=%2Fhome%2Fchi%2Fgo%2Fsrc%2Fgi
thub.com%2Fchilagrow%2FFerretDB%2Fbuild%2Fcerts%2FrootCA-cert.pem&tlsCertificateKeyFile=%2Fhome%2Fchi%2Fgo%2Fsrc%2Fgithub.com%2Fchilagrow%2FFerretDB%2Fbuild%2Fcerts%2Fclient.pem).
INFO    -test.shuffle 1721619250432656710
INFO    testing: warning: no tests to run
INFO    PASS
INFO    coverage: 2.0% of statements in ../...
INFO    coverage: 2.0% of statements in ../...
INFO    coverage: 2.0% of statements in ../...
INFO    2024-07-22T12:34:10.498+0900    INFO    debug   debug/debug.go:250      Debug server stopped.
INFO    2024-07-22T12:34:10.498+0900    INFO    observability/observability.go:113      OTLP tracer stopped     {"name":"otel"}
INFO    2024-07-22T12:34:10.514+0900    INFO    debug   debug/debug.go:250      Debug server stopped.
INFO    2024-07-22T12:34:10.515+0900    INFO    observability/observability.go:113      OTLP tracer stopped     {"name":"otel"}
INFO    2024-07-22T12:34:10.524+0900    INFO    debug   debug/debug.go:250      Debug server stopped.
INFO    2024-07-22T12:34:10.525+0900    INFO    observability/observability.go:113      OTLP tracer stopped     {"name":"otel"}
INFO    2024-07-22T12:34:10.527+0900    INFO    debug   debug/debug.go:250      Debug server stopped.
INFO    2024-07-22T12:34:10.527+0900    INFO    observability/observability.go:113      OTLP tracer stopped     {"name":"otel"}
INFO    ok      github.com/FerretDB/FerretDB/integration        1.120s  coverage: 2.0% of statements in ../... [no tests to run]
INFO    PASS github.com/FerretDB/FerretDB/integration
INFO    ok      github.com/FerretDB/FerretDB/integration/oplog  1.119s  coverage: 2.0% of statements in ../... [no tests to run]
INFO    PASS github.com/FerretDB/FerretDB/integration/oplog
INFO    ok      github.com/FerretDB/FerretDB/integration/users  1.129s  coverage: 2.0% of statements in ../... [no tests to run]
INFO    PASS github.com/FerretDB/FerretDB/integration/users
INFO    ok      github.com/FerretDB/FerretDB/integration/cursors        1.130s  coverage: 2.0% of statements in ../... [no tests to run]
INFO    PASS github.com/FerretDB/FerretDB/integration/cursors
INFO    OTLP tracer stopped
```

**Comment by chilagrow** [comment]
How envtool setup logs
```
$ task env-setup                                                                                                                                             
task: [gen-version] go generate -x ./build/version                                                                                                                                                          
go run ./generate.go                                                                                                                                                                                        
package.txt: build-host                                                                                                                                                                                     
branch.txt: slog-envtool                                                                                                                                                                                    
commit.txt: 0acd4ec75736d311a46ad3b72d68be6efc826ebf                                                                                                                                                        
version.txt: v1.21.0-103-g0acd4ec75                                                                                                                                                                         
task: [gen-version] go build -v -o bin/ -race=true -tags=ferretdb_debug,ferretdb_hana ./cmd/envtool/                                                                                                        
task: [env-setup] bin/envtool setup                                                                                                                                                                         
INFO    Waiting for 127.0.0.1:5432 to be up     {"name":"postgres"}                                                                                                                                         
INFO    Pool: connection succeed        {"username":"username"}                                                                                                                                             
INFO    Waiting for 127.0.0.1:5433 to be up     {"name":"postgres_secured"}                                                                                                                                 
INFO    Pool: connection succeed        {"username":"username"}                                                                                                                                             
INFO    Waiting for 127.0.0.1:3306 to be up     {"name":"mysql"}                                                                                                                                            
INFO    Pool: connection succeed        {"username":"root"}                                                                                                                                                 
INFO    Waiting for 127.0.0.1:47017 to be up    {"name":"mongodb"}                    
```

File: Taskfile.yml [hunk 1]
```diff
@@ -6,8 +6,8 @@ env:
   GOCOVERDIR: tmp/cover
 
 vars:
-  SHARD_INDEX: 1
-  SHARD_TOTAL: 1
+  SHARD_INDEX: 0
```
Comment (rumyantseva) [inline Taskfile.yml:9]: Why did we move it back?

**Review by rumyantseva — APPROVED** [review]
Looks good, just one question (I probably missed some context).

File: Taskfile.yml (hunk 1 above)
Comment (chilagrow) [reply Taskfile.yml:9]: What do you mean by moving it back?

Comment (rumyantseva) [reply Taskfile.yml:9]: From 1 to 0.

Comment (chilagrow) [reply Taskfile.yml:9]: https://github.com/FerretDB/FerretDB/pull/4480/files#diff-a7c8b37b0eee875782802ed67486da60bee4d47b8800407666e9c365a6715808R77

Comment (chilagrow) [reply Taskfile.yml:9]: Not allowing `--run` or `-skip` to be used at the same time with `--shard`. This simplifies the code.

Comment (AlekSi) [reply Taskfile.yml:9]: I think the real answer is that this PR also ports things from v2

**Review by noisersup — DISMISSED** [review]

**Review by AlekSi — COMMENTED** [review]
There are merge conflicts

**Review by AlekSi — APPROVED** [review]

**Review by rumyantseva — APPROVED** [review]
:shipit:

**Pull request merged**